### PR TITLE
Added support for Gemini 1.5 Flash 8B and updated available versions for Google AI and Vertex AI Gemini models

### PR DESCRIPTION
**Pull request by chrisraygill**
PR does the following:

- Adds support for Gemini 1.5 Flash 8B to the Google AI plugin
- Adds all available versions to all Gemini models in the Google AI plugin based on https://ai.google.dev/gemini-api/docs/models/gemini
- Adds all available versions to all Gemini models in the Vertex AI plugin based on https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models

Tested Gemini 1.5 Flash 8B manually in the DevUI and it works based on test apps: https://screenshot.googleplex.com/ryTsEwLtx4J2Amc

Checklist (if applicable):
- [ x] Tested (manually, unit tested, etc.)
- [ ] Docs updated

### CODE REVIEW

**Review by huangjeff5 — APPROVED** [review]

File: js/plugins/googleai/src/gemini.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ export const geminiProVision = modelRef({
 });
 
 export const gemini15Pro = modelRef({
-  name: 'googleai/gemini-1.5-pro-latest',
+  name: 'googleai/gemini-1.5-pro',
```
Comment (pavelgj) [inline js/plugins/googleai/src/gemini.ts:116]: please undo this -- breaking change

File: js/plugins/googleai/src/gemini.ts [hunk 2]
```diff
@@ -123,13 +123,18 @@ export const gemini15Pro = modelRef({
...
+      'gemini-1.5-pro-002',
+      'gemini-1.5-pro-exp-0827',
+    ],
   },
   configSchema: GeminiConfigSchema,
 });
 
 export const gemini15Flash = modelRef({
-  name: 'googleai/gemini-1.5-flash-latest',
+  name: 'googleai/gemini-1.5-flash',
```
Comment (pavelgj) [inline js/plugins/googleai/src/gemini.ts:137]: please undo this -- breaking change

File: js/plugins/googleai/src/gemini.ts [hunk 3]
```diff
@@ -172,8 +200,9 @@ export const SUPPORTED_V15_MODELS: Record<
   string,
   ModelReference<z.ZodTypeAny>
 > = {
-  'gemini-1.5-pro-latest': gemini15Pro,
-  'gemini-1.5-flash-latest': gemini15Flash,
+  'gemini-1.5-pro': gemini15Pro,
+  'gemini-1.5-flash': gemini15Flash,
```
Comment (pavelgj) [inline js/plugins/googleai/src/gemini.ts:204]: please undo this -- breaking change

**Review by pavelgj — APPROVED** [review]

File: js/plugins/googleai/src/gemini.ts [hunk 4]
```diff
@@ -139,7 +144,30 @@ export const gemini15Flash = modelRef({
...
       output: ['text', 'json'],
     },
-    versions: ['gemini-1.5-flash-001'],
+    versions: [
+      'gemini-1.5-flash',
+      'gemini-1.5-flash-001',
+      'gemini-1.5-flash-002',
+      'gemini-1.5-flash-8b-exp-0924',
+      'gemini-1.5-flash-8b-exp-0827',
+      'gemini-1.5-flash-exp-0827',
```
Comment (thatfiredev) [inline js/plugins/googleai/src/gemini.ts:153]: The [docs for experimental models](https://ai.google.dev/gemini-api/docs/models/experimental-models#use-an-experimental-model) state that:

> Support for an experimental model can change at any time.

With that in mind, should these models be marked as [unstable](https://github.com/firebase/genkit/blob/adc8ad81759f49b2791a3a87f7fa2a38d137db24/js/ai/src/model.ts#L152) in Genkit?
We'd separate them from `gemini15Flash`, `gemini15Flash8B` and `gemini15Pro`, which would only contain stable models.

```js
export const gemini15FlashExp = modelRef({
  name: 'googleai/gemini-1.5-flash-exp-0827',
  info: {
    label: 'Google AI - Gemini 1.5 Flash Experimental',
    stage: 'unstable',
    supports: {
      // (...)
    },
    versions: ['gemini-1.5-flash-exp-0827'],
  },
  configSchema: GeminiConfigSchema,
});
```

Same comment for 1.5 Flash 8B (`gemini15Flash8BExp`) and 1.5 Pro (`gemini15ProExp`).

Comment (chrisraygill) [reply js/plugins/googleai/src/gemini.ts:153]: The ways the model refs work is that the versions array is only impacting what versions are shown as available in the Developer UI, so changing these out when they lose support isn't a breaking change in the framework.

**Pull request merged**